src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -247,6 +252,45 @@ Copyright (c) .NET Foundation. All rights reserved.
       <PackProjectInputFile>$(MSBuildProjectFullPath)</PackProjectInputFile>
     </PropertyGroup>
   </Target>
+  
+  <Target Name="_GetProjectReferenceVersions"
+          Condition="'$(ResolveProjectReferenceVersionDuringPack)' == 'true'">
+    
+    <ConvertToAbsolutePath Paths="$(RestoreOutputPath)">
+      <Output TaskParameter="AbsolutePaths" PropertyName="RestoreOutputAbsolutePath" />
+    </ConvertToAbsolutePath>
+
+    <ConvertToAbsolutePath Condition="'$(ProjectAssetsFile)' != ''" Paths="$(ProjectAssetsFile)">
+      <Output TaskParameter="AbsolutePaths" PropertyName="ProjectAssetsFileAbsolutePath" />
+    </ConvertToAbsolutePath>
+
+    <GetProjectReferencesFromAssetsFileTask
+        RestoreOutputAbsolutePath="$(RestoreOutputAbsolutePath)"
+        ProjectAssetsFileAbsolutePath="$(ProjectAssetsFileAbsolutePath)">
+      <Output
+        TaskParameter="ProjectReferences"
+        ItemName="_ProjectReferencesFromAssetsFile" />
+    </GetProjectReferencesFromAssetsFileTask>
+
+    <Msbuild
+      Projects="@(_ProjectReferencesFromAssetsFile)"
+      Targets="_GetProjectVersion"
+      Properties="BuildProjectReferences=false;">
+      <Output
+        TaskParameter="TargetOutputs"
+        ItemName="_ProjectReferencesWithVersions"/>
+    </Msbuild>
+  </Target>
+
+  <Target Name="_GetProjectVersion"
+          Returns="@(_ProjectPathWithVersion)">
+    <ItemGroup>
+      <_ProjectPathWithVersion Include="$(MSBuildProjectFullPath)">
+        <ProjectVersion Condition="'$(PackageVersion)' != ''">$(PackageVersion)</ProjectVersion>
+        <ProjectVersion Condition="'$(PackageVersion)' == ''">1.0.0</ProjectVersion>
```

Comment:
would it always default to 1.0.0 or current project version?

Reply:
packageVersion already defaults to the current project version